### PR TITLE
xClusterQuorum

### PR DESCRIPTION
**Pull request by claudiospizzi**
Add **xClusterQuorum** resource and a workaround for the timing issue when deploying a two node cluster in one step. The second node does join to early because of an exception. Timeout for New-Cluster command.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xfailovercluster/11)

### CODE REVIEW

File: DSCResources/MSFT_xCluster/MSFT_xCluster.psm1 [hunk 1]
```diff
@@ -104,6 +104,8 @@ function Set-TargetResource
 
             New-Cluster -Name $Name -Node $env:COMPUTERNAME -StaticAddress $StaticIPAddress -NoStorage -Force
 
+            Start-Sleep -Seconds 60
```
Comment (TravisEz13) [inline DSCResources/MSFT_xCluster/MSFT_xCluster.psm1]: Is there a way to more explicitly waiting, rather than a sleep operation?

**Comment by TravisEz13** [comment]
Please include a summary of changes in an unreleased version change per the [contribution guidelines](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#lifecycle-of-a-pull-request)

It is required that you provide adequate test coverage for the code you change. [Contribution Guidelines -Adding test coverage for DSC resources](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#adding-test-coverage-for-dsc-resources)  Please work with us to come up with a plan to cover this code.

**Comment by TravisEz13** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_xClusterQuorum/MSFT_xClusterQuorum.psm1, line 2 \[r2\]](https://reviewable.io:443/reviews/powershell/xfailovercluster/11#-KEiJdkF1wqO89NcJaf8:-KEiJdkF1wqO89NcJaf9:-566210200) ([raw file](https://github.com/powershell/xfailovercluster/blob/4d09bfc443bd1571c302ad6506eddb1c7d27e9ac/DSCResources/MSFT_xClusterQuorum/MSFT_xClusterQuorum.psm1#L2)):*
It is required that you provide adequate test coverage for the code you change. [Contribution Guidelines -Adding test coverage for DSC resources](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#adding-test-coverage-for-dsc-resources)  Please work with us to come up with a plan to cover this code.

Suggestion: Base the tests on the [Unit tests Template](https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1)

If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.  Please consider closing and re-opening when ready for review.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11)*

**Comment by TravisEz13** [comment]
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[README.md, line 19 \[r3\]](https://reviewable.io:443/reviews/powershell/xfailovercluster/11#-KFC9-MKTz1JEhvlu3DD:-KFC9-MKTz1JEhvlu3DE:1146265129) ([raw file](https://github.com/powershell/xfailovercluster/blob/397ae0065095436106e39247157c0f5b476b88a5/README.md#L19)):*
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )
[Instructions to setup BC4, my favorite merge tool](http://www.scootersoftware.com/support.php?zz=kb_vcs#gitwindows)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11)*

**Comment by TravisEz13** [comment]
Reviewed 2 of 2 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11)*

File: DSCResources/MSFT_xCluster/MSFT_xCluster.psm1 (hunk 1 above)
Comment (itpropro) [inline DSCResources/MSFT_xCluster/MSFT_xCluster.psm1]: You could use 
While (!(Get-Cluster)){Start-Sleep 5}
This discussion is the only reason, why the Quorum Packet is still not merged. I think we should fast solve it.

**Comment by TravisEz13** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[DSCResources/MSFT_xCluster/MSFT_xCluster.psm1, line 107 \[r2\]](https://reviewable.io:443/reviews/powershell/xfailovercluster/11#-KEiJ6Q7F72yCqmibPiT-r2-107:-KLgSIOAahn5vgypypKs:1803965082) ([raw file](https://github.com/powershell/xfailovercluster/blob/4d09bfc443bd1571c302ad6506eddb1c7d27e9ac/DSCResources/MSFT_xCluster/MSFT_xCluster.psm1#L107)):*
<details><summary><i>Previously, itpropro wrote…</i></summary>
> You could use 
> 
> While (!(Get-Cluster)){Start-Sleep 5}
> 
> This discussion is the only reason, why the Quorum Packet is still not merged. I think we should solve this fast.
</details>
This seems like a much better solution

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11)*

**Comment by aultt** [comment]
@TravisEz13  it appears this PR has not made any progress forward.  Can we merge if I make the requested change as a follow up issue?  I will submit the issue now if that helps move this forward.

**Comment by TravisEz13** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/powershell/xfailovercluster/11#-KOBXSbMe2oUKYcVJnQA:-KOBXSbMe2oUKYcVJnQB:-1801949116) (no related file):*
@aultt I will merge based on your commitment to fix the open issues.  Thanks everyone!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11)*

**Comment by TravisEz13** [comment]
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/powershell/xfailovercluster/11#-KOBXSbMe2oUKYcVJnQA:-KOBY5hoipm5yqws4x3c:1927636499) (no related file):*
<details><summary><i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>
> @aultt I will merge based on your commitment to fix the open issues.  Thanks everyone!
</details>
One caveat, the merge conflicts need to be fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11)*

**Comment by mbreakey3** [comment]
@claudiospizzi if you can fix these merge conflicts we can merge these changes.

**Comment by MS-MatthewWalker** [comment]
Those options for quorum aren't correct for 2012R2 - The options are DiskWitness, FileShareWitness, NoWitness and DiskOnly the options listed in this code are correct for 2008R2 and 2012 but not later versions.  Also there is no cloudwitness in the 2016 section.

**Comment by Indhukrishna** [comment]
@claudiospizzi - Can you please fix the merge conflicts?

**Comment by claudiospizzi** [comment]
Hi all. Sorry for my long absence. I've tried to resolve the conflicts by merging the current dev branch, should be fine now.

**Comment by MS-MatthewWalker** [comment]
Ok, I see I was wrong and the correct types are being called in the set -- however this is conflicting with the actual references in the GUI and documentation as well as the independent powershell commands called.  I see this as confusing and think it should be consistent across the board.

**Comment by claudiospizzi** [comment]
Yes right, the quorum options and their naming have changed. So my implementation should cover the latest release (WS2012 R2). I've not checked WS2016 TP, because it's not released yet.

File: DSCResources/MSFT_xCluster/MSFT_xCluster.psm1 (hunk 1 above)
Comment (Indhukrishna) [inline DSCResources/MSFT_xCluster/MSFT_xCluster.psm1]: @claudiospizzi - This discussion is still unresolved. Can you please update the code to not just rely on sleep alone?

**Comment by MS-MatthewWalker** [comment]
2016 will be the same with the addition of cloud witness to the others. 

---

Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xfailovercluster/11#-:-KOoqFwzXWSw396ya-X2:b1e860d)*

**Comment by aultt** [comment]
@claudiospizzi There are still unresolved discussions before this can be merged they need to be resolved.

**Comment by Indhukrishna** [comment]
Duplicate of #22

**Pull request closed without merging**